apps/evm/lib/evm/message_call.ex

```diff
@@ -10,26 +10,52 @@ defmodule EVM.MessageCall do
     :current_machine_state,
     :current_sub_state,
     :output_params,
-    # s
     :sender,
-    # o
     :originator,
-    # r
     :recipient,
-    # c
     :code_owner,
-    # p
     :gas_price,
-    # v
     :value,
-    # v with overline
     :execution_value,
-    # d
     :data,
-    # e
-    :stack_depth
+    :stack_depth,
+    :transfer_value
```

Comment:
maybe we can pass zero for `delegatecall` instead of adding extra field?

Reply:
That's an interesting idea. So you're thinking we could just set `value` to `0` for `delegatecall`? Wouldn't that conflict with the notion that the `value` is persisted from the previous call (what lives in `exec_env.value_in_wei`)?

Reply:
Actually, you're right. I forgot that it's passed to child's exec env